source/Test.cpp

```diff
@@ -329,33 +457,65 @@ void Test::Step(Context &context, UI &menuPanels, UI &gamePanels, PlayerInfo &pl
 					if(!testData->Inject())
 						Fail(context, player, "injecting data failed");
 				}
-				++(context.stepToRun);
+				++(context.stepToRun.back());
 				break;
 			case TestStep::Type::INPUT:
-				// Give the relevant inputs here.
-				Fail(context, player, "Input not implemented");
+				if(stepToRun.command)
+				{
+					// We need to send the command through the top gamepanel, and it needs to be active.
+					if(gamePanels.IsEmpty())
+						Fail(context, player, "panel with engine not present, and can only send commands to the engine");
+					
+					if(gamePanels.Root() != gamePanels.Top())
+						Fail(context, player, "engine not active due to panel on top, and can only send commands to the engine");
+					
+					// Both get as well as the cast can result in a nullpointer. In both cases we
+					// will fail the test, since we expect the MainPanel to be here.
+					auto mainPanel = dynamic_cast<MainPanel *>(gamePanels.Root().get());
```

Comment:
Do I understand right that the goal of this dynamic_cast is to return a null pointer if `gamePanels.Root().get()` returns either null or an instance of a panel other than MainPanel? Would it be ok to ignore the wrong type failure case and use a static_cast?

I'm interested because I had been compiling [without cpp runtime support](https://emscripten.org/docs/optimizing/Optimizing-Code.html#c-rtti) as an optimization until this first dynamic_cast in the codebase. This isn't a reason to change it — compiling with Emscripten is explicitly not a goal — but if I understand right I can open a PR since it's also the preferred style in this codebase to avoid dynamic_cast.

Reply:
Ok for me to change this to a static cast.

Reply:
And if this is the only location that uses run-time type support, then feel free to open a second PR to compile without run-time support.

Reply:
Great, thanks @petervdmeer, I opened https://github.com/endless-sky/endless-sky/pull/6210 to remove.

I don't know that we want to add `-fno-rtt` and would want someone with more c++ experience to make that suggestion — but if more experienced folks want that then that'd be great!